rotkehlchen/tests/unit/test_globaldb.py

```diff
@@ -766,3 +797,94 @@ def test_add_user_owned_asset_nft(globaldb):
     new_assets = {x[0] for x in result}
     assert new_assets - initial_assets == {A_DAI.identifier, A_PICKLE.identifier, A_CRV.identifier}
     assert all(not x.startswith(NFT_DIRECTIVE) for x in new_assets)
+
+
+def test_asset_deletion(globaldb):
+    """This test checks that deletion of both evm token and normal asset works as expected"""
+    def check_tables(cursor, asset_id, expected_count, also_eth):
+        """Util function to check that data in the tables is correct. Checks that provided
+        `asset_id` either exists or doesn't exist in all tables by comparing number of found
+        entries with `expected_count` which should be either 1 or 0 respectively."""
+        queries = [
+            'SELECT * FROM assets WHERE identifier=?',
+            'SELECT * FROM common_asset_details WHERE identifier=?',
+        ]
+        if also_eth is True:
+            queries += [
+                'SELECT * FROM evm_tokens WHERE identifier=?',
+                'SELECT * FROM underlying_tokens_list WHERE parent_token_entry=?',
+            ]
+        results = []
+        for query in queries:
+            res = cursor.execute(query, (asset_id,)).fetchall()
+            results.append(len(res))
+
+        assert all([x == expected_count for x in results])
```

Comment:
Maybe move this to the utils folder in tests?

Reply:
Not sure about it since this method is not used anywhere else